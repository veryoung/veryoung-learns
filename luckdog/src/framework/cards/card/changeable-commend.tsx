/* eslint-disable @typescript-eslint/prefer-for-of */
import React from 'react';
import { View, Text, StyleSheet } from '@tencent/hippy-react-qb';
import { CommonProps } from '../../../entity';
import { BeaconReportProps, reportUDS, BusiKey, strictExposeReporter } from '@/luckdog';
import FeedsProtect from '../../../mixins/FeedsProtect';
import { FeedsLineHeight, FeedsTheme, FeedsUIStyle } from '../../../feeds-styles/tab-22/components/utils';
import { scale } from '../../../components/animationStyle';
import { CommonCardStyle, SmallBookCoverLeftTagStyle } from '../../FeedsConst';
import { Title } from '../components/title';
import { RefreshCard, CardLayout } from '../../protocol/card';
import { shouldComponentUpdate } from '@/luckbox';
import FeedsUtils from '../../FeedsUtils';
import { BookCoverRadiusStyle } from '../../../types/card';
import { BookCover } from '../../../components/book-font-cover';
import { BaseBook, UIType } from '../../protocol';

interface Props extends BeaconReportProps, CommonProps {
  index: number;
  key: string;
  globalConf: any;
  totalLength: number;
  curTabId: number;
  data: RefreshCard;
}
// 默认一屏展示的书籍数
// const DEFAULT_PER_NUMBER = 8;

const { width } = FeedsUtils.getScreen();
const IMAGE_WIDTH = (width - (12 * 2) - (16 * 2) - (26 * 3)) / 4;
const IMAGE_HEIGHT = (IMAGE_WIDTH * 81) / 60;

interface State {
  /** 页面所在位置 */
  position: number;
  /** 当前页书籍内容 */
  books: [];
}

/** 支持可换一换多行推荐卡 */
@FeedsProtect.protect
export class ChangeableComment extends React.Component<Props> {
  public static getRowType() {
    return UIType.CHANGEABLE_COMMEND;
  }
  /** 点击动画效果 */
  public scaleAnim = scale(200, 1.015);
  /** 当前页面状态 */
  public state: State = {
    position: 0,
    books: [],
  };

  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'changeableCommend');
  }

  /** 计算布局 */
  public onTitleLayout = (event) => {
    strictExposeReporter.updateTitleHeight(this.props.index, event.layout.height);
  };

  /**
   * 切换猜你想看 换一换
   */
  public switchNovel = () => {
    const { position } = this.state;
    const { data } = this.props;
    const { pageList } = data;
    const pageIndex = pageList?.length - 1 || 0;
    let newPosition = position;
    if (pageIndex > 0) {
      newPosition = newPosition >= pageIndex ? 0 : newPosition += 1;
    }
    this.setState({
      position: newPosition,
    });

    this.reportUDS(BusiKey.CLICK__CARD_CHANGE);
  };

  public onBookItemLayout = (event, tabIndex, bookIndex, bookId, rowIndex) => {
    const { height } = event.layout;
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.index,
      tabIndex,
      bookIndex,
      bookId,
      rect: {
        ...event.layout,
        y: (height + styles.item.marginBottom) * rowIndex,
      },
    });
  };

  /**
   * 点击书籍
   * @param {*} book 书籍
   * @param {*} index 书籍下标
   */
  public handleClick = (book) => {
    const { resourceId = '', jumpUrl } = book || {};
    FeedsUtils.doLoadUrl(jumpUrl, `${this.props.curTabId}`);
    this.reportUDS(BusiKey.CLICK__CARD_BOOK, { book_id: resourceId });
  };

  /** 按照页码渲染页面内容 */
  public renderPage = () => {
    const view: any[] = [];
    const { data, index: cardIndex } = this.props;
    const { pageList, isRow } = data;
    const { position } = this.state;
    const currentPage = pageList[position] || [];

    const bookIds = currentPage.reduce((acc: string[], page): string[] => {
      const ids = page.dataList.map(({ resourceId }) => resourceId);
      return [...acc, ...ids];
    }, []);
    strictExposeReporter.updateBookIds(cardIndex, 0, bookIds);

    for (let i = 0; i < currentPage.length; i++) {
      const cardLayout = currentPage[i];
      const bookIndexBase = (currentPage[i - 1]?.dataList?.length || 0) * i;
      view.push(<View
        key={`${position}_${i}`}
      >
        {
          this.renderBook(cardLayout, bookIndexBase, i)
        }
      </View>);
    }

    return <View
      style={{
        flexDirection: isRow ? 'row' : 'column',
      }}
    >
      {view}
    </View>;
  };

  /** 渲染具体页面内容 */
  public renderBook = (cardLayout: CardLayout<BaseBook>, bookIndexBase: number, rowIndex: number) => {
    const { dataList, isRow } = cardLayout;
    const view: any[] = [];
    for (let i = 0; i < dataList.length; i++) {
      const book = dataList[i];
      const { resourceId, score, picUrl = '', tag, resourceName } = book;
      let leftTag: any = undefined;
      if (tag) {
        leftTag = FeedsUtils.getBookTagStyle(tag, SmallBookCoverLeftTagStyle);
      }
      view.push(<View
        style={styles.item}
        key={resourceId}
        onLayout={e => this.onBookItemLayout(e, 0, bookIndexBase + i, resourceId, rowIndex)}
        onClick={strictExposeReporter.triggerExpoCheck(() => this.handleClick(book))}
      >
        <View>
          <BookCover
            height={IMAGE_HEIGHT}
            width={IMAGE_WIDTH}
            url={picUrl}
            bookID={resourceId}
            radius={BookCoverRadiusStyle.SMALL}
            sourceFrom={resourceId}
            leftTag={leftTag}
          />
          <Text style={styles.bookName} numberOfLines={2}>
            {resourceName}
          </Text>
        </View>
        {
          score ? <View style={styles.textScoreBlock}>
            <Text style={styles.textScore}>{score}</Text>
          </View> : null
        }
      </View>);
    }
    return <View
      style={{
        flexDirection: isRow ? 'row' : 'column',
      }}
    >
      {view}
    </View>;
  };


  public render() {
    const { data } = this.props;
    const { title, pageList } = data;
    if (!pageList || pageList.length === 0) return null;

    return (
      <View style={[{ ...CommonCardStyle }, { transform: [{ scale: this.scaleAnim }] }]}>
        <View collapsable={false}>
          <Title
            title={title}
            onLayout={this.onTitleLayout}
            changeable={true}
            switchNovel={this.switchNovel}
            parent={this}
          />
        </View>
        <View style={styles.wrap}>
          {
            this.renderPage()
          }
        </View>
      </View>
    );
  }

  private reportUDS = (eventKey: BusiKey, moreData = {}) => {
    const { reportInfo, uiType } = this.props.data || {};
    reportUDS(eventKey, {}, { ...reportInfo, bigdata_contentid: '', ui_type: uiType, ...moreData });
  };
}

const styles = StyleSheet.create({
  wrap: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingLeft: 3,
    paddingRight: 3,
  },

  item: {
    marginLeft: 13,
    marginRight: 13,
    width: IMAGE_WIDTH,
    marginBottom: 24,
    justifyContent: 'space-between',
  },
  textScoreBlock: {
    flexDirection: 'row',
  },
  textScore: {
    fontFamily: 'PingFangSC-Semibold',
    fontSize: FeedsUIStyle.T1,
    colors: FeedsTheme.SkinColor.N3,
    fontWeight: 'bold',
    lineHeight: FeedsLineHeight.T1,
  },
  bookName: {
    colors: FeedsTheme.SkinColor.A1,
    fontSize: FeedsUIStyle.T1,
    lineHeight: FeedsLineHeight.T1,
    marginBottom: 2,
    marginTop: 4,
  },
});


