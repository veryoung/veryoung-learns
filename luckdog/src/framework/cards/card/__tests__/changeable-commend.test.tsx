import React from 'react';
import renderer from 'react-test-renderer';

import { ChangeableComment } from '../changeable-commend';
import { changeableCommendData } from './mocks/changeable-commend';

describe('<ChangeableComment/>', () => {
  describe('render testing', () => {
    it('render changeable commend', () => {
      const props = {
        data: changeableCommendData as any,
        index: 3,
        key: 'test',
        globalConf: {},
        totalLength: 32,
        curTabId: 181,
      };
      const tree = renderer.create(<ChangeableComment {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
