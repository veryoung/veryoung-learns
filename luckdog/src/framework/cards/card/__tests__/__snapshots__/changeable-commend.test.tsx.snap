// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeableComment/> render testing render changeable commend 1`] = `
<div
  nativeName="View"
  style={
    Array [
      Object {
        "backgroundColors": Array [
          "white",
          "#23282d",
          "white",
          "rgba(0, 0, 0, 0.8)",
        ],
        "borderRadius": 12,
        "marginBottom": 0,
        "marginHorizontal": 12,
        "marginTop": 16,
      },
      Object {
        "transform": Array [
          Object {
            "scale": AnimationSet {
              "animationId": undefined,
              "animationList": Array [
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
              ],
              "onHippyAnimationCancel": [Function],
              "onHippyAnimationEnd": [Function],
              "onHippyAnimationRepeat": [Function],
              "onHippyAnimationStart": [Function],
              "onRNfqbAnimationCancel": [Function],
              "onRNfqbAnimationEnd": [Function],
              "onRNfqbAnimationRepeat": [Function],
              "onRNfqbAnimationStart": [Function],
            },
          },
        ],
      },
    ]
  }
>
  <div
    nativeName="View"
    style={
      Object {
        "collapsable": false,
      }
    }
  >
    <div
      nativeName="View"
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Array [
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "maxWidth": 252,
              "paddingBottom": 14,
              "paddingTop": 14,
            },
            Object {},
          ]
        }
        text="玄幻推荐"
      >
        玄幻推荐
      </p>
      <div
        className="change-view"
        nativeName="View"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "paddingBottom": 14,
            "paddingTop": 14,
            "position": "relative",
          }
        }
      >
        <p
          nativeName="Text"
          numberOfLines={1}
          style={
            Array [
              Object {
                "colors": Array [
                  "#000000",
                  "#686D74",
                  "#000000",
                  "#ffffff",
                ],
                "fontSize": 12,
                "lineHeight": 16,
              },
              Object {},
            ]
          }
          text="换一换"
        >
          换一换
        </p>
      </div>
    </div>
  </div>
  <div
    nativeName="View"
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingLeft": 3,
        "paddingRight": 3,
      }
    }
  >
    <div
      nativeName="View"
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1126435253",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F253%2F26435253%2Ft3_26435253.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="雪中悍刀行"
              >
                雪中悍刀行
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.7分"
              >
                9.7分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1100462521",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F521%2F462521%2Ft3_462521.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
                <img
                  alt=""
                  nativeName="Image"
                  src="https://today.imtt.qq.com/novelsingletab/badge/big_original.png"
                  style={
                    Object {
                      "0": Object {
                        "position": "absolute",
                        "right": 3,
                        "top": 3,
                      },
                      "1": Object {
                        "height": 24,
                        "left": 6,
                        "top": 0,
                        "width": 14,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="完美世界"
              >
                完美世界
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.8分"
              >
                9.8分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1132786211",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F211%2F32786211%2Ft3_32786211.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="一剑独尊"
              >
                一剑独尊
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.8分"
              >
                9.8分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1100468021",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F21%2F468021%2Ft3_468021.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
                <img
                  alt=""
                  nativeName="Image"
                  src="https://today.imtt.qq.com/novelsingletab/badge/big_original.png"
                  style={
                    Object {
                      "0": Object {
                        "position": "absolute",
                        "right": 3,
                        "top": 3,
                      },
                      "1": Object {
                        "height": 24,
                        "left": 6,
                        "top": 0,
                        "width": 14,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="斗罗大陆"
              >
                斗罗大陆
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.6分"
              >
                9.6分
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1126573913",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F913%2F26573913%2Ft3_26573913.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="无敌剑域"
              >
                无敌剑域
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.6分"
              >
                9.6分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1132840061",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F61%2F32840061%2Ft3_32840061.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="逆天邪神"
              >
                逆天邪神
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.7分"
              >
                9.7分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1121058239",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F239%2F21058239%2Ft3_21058239.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
                <img
                  alt=""
                  nativeName="Image"
                  src="https://today.imtt.qq.com/novelsingletab/badge/big_onlyfree.png"
                  style={
                    Object {
                      "0": Object {
                        "position": "absolute",
                        "right": 3,
                        "top": 3,
                      },
                      "1": Object {
                        "height": 24,
                        "left": 6,
                        "top": 0,
                        "width": 14,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="伏天氏"
              >
                伏天氏
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.7分"
              >
                9.7分
              </p>
            </div>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "justifyContent": "space-between",
                "marginBottom": 24,
                "marginLeft": 13,
                "marginRight": 13,
                "width": 56.5,
              }
            }
          >
            <div
              nativeName="View"
              style={Object {}}
            >
              <div
                nativeName="View"
                style={
                  Array [
                    Object {
                      "position": "relative",
                    },
                    Object {
                      "height": 76.275,
                      "width": 56.5,
                    },
                  ]
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  reportData={
                    Object {
                      "sourceFrom": "1100462596",
                    }
                  }
                  src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F596%2F462596%2Ft3_462596.jpg"
                  style={
                    Object {
                      "0": Object {
                        "borderRadius": 6,
                        "height": 76.275,
                        "width": 56.5,
                      },
                      "1": Object {
                        "borderColors": Array [
                          "#e5e5e5",
                          "#31363b",
                          "#00000010",
                          "#ffffff10",
                        ],
                        "borderWidth": 0.5,
                      },
                    }
                  }
                />
                <img
                  alt=""
                  nativeName="Image"
                  src="https://today.imtt.qq.com/novelsingletab/badge/big_onlyfree.png"
                  style={
                    Object {
                      "0": Object {
                        "position": "absolute",
                        "right": 3,
                        "top": 3,
                      },
                      "1": Object {
                        "height": 24,
                        "left": 6,
                        "top": 0,
                        "width": 14,
                      },
                    }
                  }
                />
              </div>
              <p
                nativeName="Text"
                numberOfLines={2}
                style={
                  Object {
                    "colors": Array [
                      "#242424",
                      "#686d74",
                      "#242424",
                      "white",
                    ],
                    "fontSize": 12,
                    "lineHeight": 16,
                    "marginBottom": 2,
                    "marginTop": 4,
                  }
                }
                text="武极天下"
              >
                武极天下
              </p>
            </div>
            <div
              nativeName="View"
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <p
                nativeName="Text"
                style={
                  Object {
                    "colors": Array [
                      "#FF796E",
                      "#794543",
                      "#FF796E",
                      "#794543",
                    ],
                    "fontFamily": "PingFangSC-Semibold",
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "lineHeight": 16,
                  }
                }
                text="9.5分"
              >
                9.5分
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
